src/auto/classify.rs

```diff
@@ -1,52 +1,57 @@
 use std::error::Error;
 
-use serde::{Serialize, Deserialize};
+use serde::{Deserialize, Serialize};
 
-use crate::{ProgramInfo, Message};
+use crate::{Message, ProgramInfo};
 
 use super::try_parse_json;
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Classification {
     #[serde(rename = "thoughts on how to classify it")]
     thoughts: String,
-    
+
     #[serde(rename = "message classification")]
     classification: String,
 }
 
 pub fn is_task(program: &mut ProgramInfo, task: &str) -> Result<bool, Box<dyn Error>> {
-    let ProgramInfo { 
-        context, .. 
-    } = program;
+    let ProgramInfo { context, .. } = program;
     let mut context = context.lock().unwrap();
-    
+
     context.agents.fast.llm.prompt.clear();
     context.agents.fast.llm.message_history.clear();
-    
-    context.agents.fast.llm.prompt.push(Message::Assistant(format!(r#"
-Given a message respond with one of the following.
-
-"conversational": A conversational message
-"task": A task or request
-"#)));
-
 
-    context.agents.fast.llm.message_history.push(Message::User(format!(r#"
-Respond in this format:
-
-```json
+    context
+        .agents
+        .fast
+        .llm
+        .prompt
+        .push(Message::Assistant(String::from(
+            "Respond with either 'conversational' for a chat message or 'task' for a request.",
+        )));
+
+    context
+        .agents
+        .fast
+        .llm
+        .message_history
+        .push(Message::User(String::from(
+            r#"Respond in JSON:
 {{
-    "thoughts on how to classify it": "...",
-    "message classification": "..."
-}}
-```"#)));
-
-    context.agents.fast.llm.message_history.push(Message::User(format!(
-        "Request to Classify: {task}"
-    )));
+"thoughts on how to classify it": "<thoughts_here>",
+"message classification": "<classification_here>"
+}}"#,
+        )));
```

Comment:
limited test shows that this shortened version is enough to make llm produce consistent output as original version